jsx/DataTable.js

```diff
@@ -331,7 +331,7 @@ class DataTable extends Component {
   }
 
   render() {
-    if (this.props.data === null || this.props.data.length === 0) {
+    if ((this.props.data === null || this.props.data.length === 0) && this.props.nullTableShow !== 'show' ) {
```

Comment:
It seems that the `if` statement will only be executed when `this.props.data === null ` && this.props.data.length !== 0:
because: `this.props.data.length === 0` traced back to to line 198 `const treeTable = (this.state.tableData.length === 0)?`  in `modules/document_repository/jsx/docIndex.js`, at that same place, the `nullTableShow="show"` is defined. and it is passed down to here.
This means, if `this.props.data.length === 0`, then `this.props.nullTableShow === 'show'`

Reply:
@coreyAbout 
Before If the data is empty(this.props.data === null or this.props.data.length === 0), this will render a null table. In this particular case. When data is empty, we don't want the render a null table, that is why I put a props variable here, and this props variable will not bother the other modules with react table. "this.props.nullTableShow !== 'show'" always is true, if you don't give a value to nullTableShow. [This means, if this.props.data.length === 0, then this.props.nullTableShow === 'show'] that is what I extractly want. It means don't render this part, jump out this if condition.

Reply:
@kongtiaowang are you still going to add the change that we talked about on tuesday that if this.props.childrenNode is not empty then make this.props.nullTableShow = 'show'?

Reply:
@kongtiaowang 
Thanks for the explanation! I think we may leave a little comment in the code. I have no other problems. Cheers!

Reply:
A comment may be something like (you choose the right word): 
this.props.nullTableShow !== 'show' is for: do not render this if statement when there is null data for doc_repository, where nullTableShow is set to show and passed down here.

Reply:
@zaliqarosli I tried, it is will cause the React Datatable failed. The Datatable only accept the data one time. If you change the table or anything in this table it will crash. That is why I used two tables (Datatable and nullDatatable) to render this module.